gossipd/routing.c

```diff
@@ -988,7 +988,7 @@ find_shorter_route(const tal_t *ctx, struct routing_state *rstate,
 		   struct amount_msat *fee)
 {
 	struct unvisited *unvisited;
-	struct chan **short_route;
+	struct chan **short_route = NULL;
```

Comment:
Compiler is right in this case --- We could `goto bad_total` which goes through the `tal_free(short_route)` before the `assert(short_route)`.